### DISCUSSION TITLE
mitmproxy fro alpine image

### DISCUSSION THREAD

**Discussion by ghost** [mitmproxy/mitmproxy#6427] [General]
I can see that there is a Alpine package for mitmproxy in the testing repostory. Is there any way that this package becomes stable in can be used in production?

**Comment by mhils** [comment]
From https://docs.mitmproxy.org/stable/overview-installation/#linux:

> Some Linux distributions provide community-supported mitmproxy packages through their native package repositories (e.g., Arch Linux, Debian, Ubuntu, Kali Linux, OpenSUSE, etc.). We are not involved in the maintenance of downstream packaging efforts, and they often lag behind the current mitmproxy release. Please contact the repository maintainers directly for issues with native packages.

For containers, the only thing we recommend (and support) at the moment is https://hub.docker.com/r/mitmproxy/mitmproxy :)

**Reply by ghost** [reply]
Thanks for your response. 
I sounds that a stable alpine package is currently not planned. In the environment I maintain it is required to don't have any critical vulnerabilities. And this official image you has two.


$ trivy image mitmproxy/mitmproxy:latest
2023-10-30T12:49:20.857+0100	INFO	Vulnerability scanning is enabled
2023-10-30T12:49:20.857+0100	INFO	Secret scanning is enabled
2023-10-30T12:49:20.857+0100	INFO	If your scanning is slow, please try '--scanners vuln' to disable secret scanning
2023-10-30T12:49:20.857+0100	INFO	Please see also https://aquasecurity.github.io/trivy/v0.44/docs/scanner/secret/#recommendation for faster secret detection
2023-10-30T12:49:20.865+0100	INFO	Detected OS: debian
2023-10-30T12:49:20.865+0100	INFO	Detecting Debian vulnerabilities...
2023-10-30T12:49:20.873+0100	INFO	Number of language-specific files: 1
2023-10-30T12:49:20.873+0100	INFO	Detecting python-pkg vulnerabilities...

mitmproxy/mitmproxy:latest (debian 11.7)

Total: 109 (UNKNOWN: 0, LOW: 72, MEDIUM: 18, HIGH: 17, CRITICAL: 2)

**Reply by mhils** [reply]
I understand where you are coming from, but unfortunately we don't have the resources to provide Debian/Ubuntu/Kali/OpenSUSE/Alpine/... images. We provide PyPI packages as well if that helps.

The critical vulnerabilities your tool is reporting are not reachable/used when running mitmproxy, so at this point it's a compliance concern and not a security concern. Of course, we're happy to set up a support contract and provide SLAs for these sorts of issues. Email is on my profile. :-)

**Reply by ghost** [reply]
Via pip i tried it before but it doesn't work:

`
error: subprocess-exited-with-error

× Preparing metadata (pyproject.toml) did not run successfully.
│ exit code: 1
╰─> [6 lines of output]

  Cargo, the Rust package manager, is not installed or is not on PATH.
  This package requires Rust and Cargo to compile extensions. Install it through
  the system's package manager or via https://rustup.rs/
  
  Checking for Rust toolchain....
  [end of output]

note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.
`
I used pip and not pipx because pipx wouln't work without bash.
I used the docker image python:3.12-alpine3.18

**Reply by mhils** [reply]
Cargo, the Rust package manager, is not installed or is not on PATH.
This package requires Rust and Cargo to compile extensions. Install it through
the system's package manager or via https://rustup.rs/

**Reply by ghost** [reply]
thanks this builds new issues as well. Will work on it thanks!
